sem/tree: remove IndexedVarContainer format method

mgartner opened the pull request:
#### sem/tree: remove IndexedVarContainer format method

`IndexedVarContainer.IndexedVarNodeFormatter` is a vestigial interface
method that is no longer needed. All indexed vars can be formatted in
the canonical representation: "@%d". This cuts the size of the
`IndexedVar` struct in half.

Informs #117546

Release note: None

#### sql: remove implementations of eval.IndexedVarContainer

This commit removes some unnecessary implementations of
`eval.IndexedVarContainer`. It also renames
`tree.MakeTypesOnlyIndexedVarHelper` to
`tree.MakeIndexedVarHelperWithTypes.

Release note: None

#### sem/tree: remove fmtSymbolicVars format flag

The `fmtSymbolicVars` format flag is no longer used and has been
removed.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/118684)
<!-- Reviewable:end -->

mgartner committed:
sem/tree: remove IndexedVarContainer format method

`IndexedVarContainer.IndexedVarNodeFormatter` is a vestigial interface
method that is no longer needed. All indexed vars can be formatted in
the canonical representation: "@%d". This cuts the size of the
`IndexedVar` struct in half.

Release note: None

mgartner committed:
sql: remove implementations of eval.IndexedVarContainer

This commit removes some unnecessary implementations of
`eval.IndexedVarContainer`. It also renames
`tree.MakeTypesOnlyIndexedVarHelper` to
`tree.MakeIndexedVarHelperWithTypes.

Release note: None

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 15 of 15 files at r1, 8 of 8 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/118684#-:-NpuNldNCx-WPHlm0Y8t:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @DrewKimball and @mgartner)

___
*[`pkg/sql/sem/tree/indexed_vars.go` line 84 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/118684#-NpuN19rA1Qf_QXQI9dz:-NpuN19rA1Qf_QXQI9e-:b-7jqy6w) ([raw file](https://github.com/cockroachdb/cockroach/blob/15961a19faca0e2b66df2d01a547549523ca70c7/pkg/sql/sem/tree/indexed_vars.go#L84)):*
> ```Go
> 	if ctx.indexedVarFormat != nil {
> 		ctx.indexedVarFormat(ctx, v.Idx)
> 	} else if f.HasFlags(fmtSymbolicVars) || v.col == nil {
> ```

I think we can remove `fmtSymbolicVars`, right?


<!-- Sent from Reviewable.io -->

mgartner commented:
>I think we can remove fmtSymbolicVars, right?

Oh yes! Great catch. I meant to do this but had forgotten. Will push another commit shortly.

mgartner committed:
sem/tree: remove fmtSymbolicVars format flag

The `fmtSymbolicVars` format flag is no longer used and has been
removed.

Release note: None

DrewKimball reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 15 of 15 files at r1, 8 of 8 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/118684#-:-NpunnHBDfwC84UkM6_M:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @mgartner)



<!-- Sent from Reviewable.io -->
